plugins/processors/starlark/testdata/sparkplug.star

```diff
@@ -0,0 +1,305 @@
+    #   source, datatype, and metric name
+    # Extract these fields and include them as tags.
+    fields = name.split('/')
+    nfields = len(fields)
+    if nfields > 0: 
+        metric.tags["Source"] = fields[0]
+        if nfields > 1:
+            # Shorten excessively long datatype strings
+            metric.tags["Datatype"] = fields[1]
+            if nfields > 2: 
+                metric.tags["Metric"] = fields[2]
+
+                # OPTIONAL
+                #
+                # By using underscore characters the metric name can be further
+                # divided into additional tags.  
+                # How this is defined is site specific.  
+                # Customize this as you wish
+
+                # The following demonstrates dividing the metric name into 4 new tags
+                # A metric name can have from 2-5 underscore separated fields 
+                
+                # If there is only one field then the only tag created is 'metric'
+                #
+                # The last field is Units is filled before fields 3, 4 and 5
+                # Ex: C, V, Torr, W, psi, RPM, On....
+                #
+                # 
+                # Fields 3, 4 and 5 (device, position, composition) are optional
+                #    measurement_component_device_position_composition_units
+                #
+                # Ex:  I_FuelTank1_C                    (2 fields) 
+                #         Measurement   I
+                #         Component     FuelTank1   
+                #         Units         C
+                #      I_FuelTank1_TC_Outlet_C          (5 fields)           
+                #         Measurement   I
+                #         Component     FuelTank1   
+                #         Device        TC
+                #         Position      Outlet
+                #         Units         C
+                #      I_FuelTank1_TC_Outlet_Premium_C  (6 fields) 
+                #         Measurement   I
+                #         Component     FuelTank1   
+                #         Device        TC  
+                #         Position      Outlet
+                #         Composition   Premium   
+                #         Units         C
+
+                # Split the metric name into fields using '_' 
+                sfields = fields[2].split('_')
+                nf = len(sfields)
+                # Don't split the name if it's one or two fields 
+                if nf <= 2:
+                    metric.name = "Name"
+                else:
+                    metric.name = sfields[nf-1]     # The Units are used for the metric name
+                    metric.tags["Component"] = sfields[1]
+                if nf > 3:
+                    metric.tags["Device"] = sfields[2]
+                if nf > 4:
+                    metric.tags["Position"] = sfields[3]
+                if nf > 5:
+                    metric.tags["Composition"] = sfields[4]
```

Comment:
Please reduce indention by one level. I know it costs some more cycles (as we need to check 2 conditions) but the readability is (especially for that example) more important.